nsls2forge_utils/auto_tick.py

```diff
@@ -0,0 +1,627 @@
+                            eval_cmd(f"rm -rf {f}")
+
+    if not dry_run:
+        logger.info(
+            "API Calls Remaining: %d",
+            mctx.gh.rate_limit()["resources"]["core"]["remaining"],
+        )
+    logger.info("Done")
+
+
+def status_report():
+    '''
+    Write out the status of current/recent migrations and their
+    pull requests on GitHub.
+
+    Only works for Version migrations at the moment.
+    '''
+    print('Determining current status of migrations...')
+    mctx, *_, migrators = initialize_migrators()
+    if not os.path.exists("./status"):
+        os.mkdir("./status")
+
+    for migrator in migrators:
+        if isinstance(migrator, Version):
+            write_version_migrator_status(migrator, mctx)
+
+    lst = [
+        k
+        for k, v in mctx.graph.nodes.items()
+        if len(
+            [
+                z
+                for z in v.get("payload", {}).get("PRed", [])
+                if z.get("PR", {}).get("state", "closed") == "open"
+                and z.get("data", {}).get("migrator_name", "") == "Version"
+            ],
+        )
+        >= Version.max_num_prs
+    ]
+    with open("./status/could_use_help.json", "w") as f:
+        json.dump(
+            sorted(
+                lst,
+                key=lambda z: (len(nx.descendants(mctx.graph, z)), lst),
+                reverse=True,
+            ),
+            f,
+            indent=2,
+        )
+    print('Statuses have been placed in ./status')
+
+
+def clean(include=None, exclude=None, yes=False):
+    '''
+    Cleans the current directory for a fresh run of the
+    graph building and auto-tick process.
+
+    Parameters
+    ----------
+    include: list, optional
+        Files to be removed (default is all files required
+        by the bot except names.txt)
+    exclude: list, optional
+        Files to keep if already included either by default
+        or by user
+    yes: bool, optional
+        Automatically proceed with deletion if True
+    '''
+    to_be_removed = [
+        './dask-worker-space/*',
+        './node_attrs/*',
+        './feedstocks/*',
+        './pr_json/*',
+        './versions/*',
+        './status/*',
+        'graph.json'
+    ]
+    if include is not None:
+        to_be_removed += include
+    if exclude is not None:
+        to_be_removed = [p for p in to_be_removed if p not in exclude]
+    rm_string = '\n'.join(to_be_removed)
+    warn_msg = (
+        'WARNING: This will delete all files associated with creating the graph.\n'
+        'This includes:\n\n'
+        f'{rm_string}\n\n'
+        'If you would like to exclude some file that is included, please run:\n'
+        'auto-tick clean --exclude FILE1 FILE2\n\n'
+        'If you would like to include some file that is excluded, please run:\n'
+        'auto-tick clean --include FILE1 FILE2\n\n'
+        'NOTE: If the same file exists in both include and exclude it will NOT be removed.'
+    )
```

Comment:
Tripple quotes will allow you to avoid `\n`s and it can be formatted the same way as rendered.

Reply:
I was having an issue in my editor where it would render the indentation as spaces beforehand, so I decided to use single quotes instead. Otherwise the checks wouldn't pass flake8.

Reply:
OK, no worries. It was a small style suggestion.